apps/src/codebridge/InfoPanel/ValidationResults.tsx

```diff
@@ -0,0 +1,98 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import codebridgeI18n from '@cdo/apps/codebridge/locale';
+import {ValidationResult} from '@cdo/apps/lab2/progress/ProgressManager';
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import moduleStyles from './styles/validation-results.module.scss';
+
+interface ValidationResultsProps {
+  className?: string;
+}
+
+const ValidationResults: React.FunctionComponent<ValidationResultsProps> = ({
+  className,
+}) => {
+  const {validationResults, satisfied} = useAppSelector(
+    state => state.lab.validationState
+  );
+  const isTesting = useAppSelector(state => state.lab2System.isTesting);
+
+  if (!validationResults) {
+    return null;
+  }
+
+  function getClassForResult(result: ValidationResult) {
+    switch (result.result) {
+      case 'PASS':
+      case 'EXPECTED_FAILURE':
+        return classNames(moduleStyles.passIcon, 'fa-solid fa-check-circle');
+      case 'FAIL':
+      case 'UNEXPECTED_SUCCESS':
+        return classNames(moduleStyles.failIcon, 'fa-solid fa-times-circle');
+      case 'SKIP':
+        return classNames(moduleStyles.cautionIcon, 'fa-solid fa-minus-circle');
+      case 'ERROR':
+        return classNames(
+          moduleStyles.failIcon,
+          'fa-solid fa-exclamation-circle'
+        );
+    }
+  }
```

Comment:
nit, but could this be lifted up outside of the component definition? 

I wouldn't break it out to a separate file unless you think it'd be generically useful.

Reply:
done!